### PR TITLE
feat(sdk-coin-xrp): add token transaction builders

### PR DESCRIPTION
**Pull request by nvrakesh06**
Ticket: [WIN-3610](https://bitgoinc.atlassian.net/browse/WIN-3610)

[WIN-3610]: https://bitgoinc.atlassian.net/browse/WIN-3610?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[feat(sdk-](https://github.com/BitGo/BitGoJS/pull/5022/commits/b560dfa574d6a29a303ff9136cdbbd8931d8dd6b)[coin-xrp](https://bitgoinc.atlassian.net/browse/COIN-xrp)[): add tokenTransferBuilder](https://github.com/BitGo/BitGoJS/pull/5022/commits/b560dfa574d6a29a303ff9136cdbbd8931d8dd6b)
[feat(sdk-](https://github.com/BitGo/BitGoJS/pull/5022/commits/f8bff1d0509a8b69072cdac18eba6f50c5d40493)[coin-xrp](https://bitgoinc.atlassian.net/browse/COIN-xrp)[): add trustsetBuilder](https://github.com/BitGo/BitGoJS/pull/5022/commits/f8bff1d0509a8b69072cdac18eba6f50c5d40493)

### CODE REVIEW

File: modules/sdk-coin-xrp/src/lib/transferBuilder.ts [hunk 1]
```diff
@@ -56,7 +55,7 @@ export class TransferBuilder extends TransactionBuilder {
    * @param {string} amount - the amount sent
    * @returns {TransactionBuilder} This transaction builder
    */
-  amount(amount: string): TransactionBuilder {
+  amount(amount: Amount): TransactionBuilder {
     if (typeof amount !== 'string') {
       throw new Error(`amount type ${typeof amount} must be a string`);
     }
```
Comment (kamleshmugdiya) [inline modules/sdk-coin-xrp/src/lib/transferBuilder.ts:61]: this type check will always fail.

**Review by kamleshmugdiya — DISMISSED** [review]

File: modules/sdk-coin-xrp/src/lib/transferBuilder.ts (hunk 1 above)
Comment (kamleshmugdiya) [reply modules/sdk-coin-xrp/src/lib/transferBuilder.ts:61]: my bad, one of type of Amount is string.

**Comment by kamleshmugdiya** [comment]
is it possible to add tests?

File: modules/sdk-coin-xrp/src/lib/tokenTransferBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        !_.isString(amount) &&
+        _.isObjectLike(amount) &&
+        _.isString(amount.currency) &&
+        _.isString(amount.issuer) &&
+        _.isString(amount.value)
+      );
+    }
+
+    if (!isIssuedCurrencyAmount(amount)) {
+      throw new Error(`amount type ${typeof amount} must be a IssuedCurrencyAmount type`);
```
Comment (dpkjnr) [inline modules/sdk-coin-xrp/src/lib/tokenTransferBuilder.ts:72]: this message looks confusing, what is IssuedCurrencyAmount

File: modules/sdk-coin-xrp/src/lib/transaction.ts [hunk 3]
```diff
@@ -51,6 +51,16 @@ export class Transaction extends BaseTransaction {
     return true;
   }
 
+  private isIssuedCurrencyAmount(amount: Amount): amount is IssuedCurrencyAmount {
```
Comment (dpkjnr) [inline modules/sdk-coin-xrp/src/lib/transaction.ts:54]: looks duplicate, maybe move it to utils and use at both the places

**Review by dpkjnr — COMMENTED** [review]
Added a few comments, can you also add unit tests for the new type of txn both at txn level as well as transactionBuilder level

**Comment by nvrakesh06** [comment]
I have tested the builder using bitgojs-alpha release and hitansh reviewed it.
I have created a [ticket](https://bitgoinc.atlassian.net/browse/WIN-3683). Will add unit test in later PRs.

File: modules/sdk-core/src/account-lib/baseCoin/enum.ts [hunk 4]
```diff
@@ -70,6 +70,8 @@ export enum TransactionType {
   // Closing an associated token account (e.g. SOL)
   CloseAssociatedTokenAccount,
   SingleNominatorWithdraw,
+  SendToken,
+  TrustLine,
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/account-lib/baseCoin/enum.ts:74]: Please add a description, like we do for the other types.

**Review by zahin-mohammad — DISMISSED** [review]

File: modules/sdk-coin-xrp/src/lib/utils.ts [hunk 5]
```diff
@@ -210,6 +218,37 @@ class Utils implements BaseUtils {
...
+   * Get the associated XRP Currency details from token name. Throws an error if token is unsupported
+   * @param {string} tokenName - The token name
+   */
+  public getXrpCurrencyFromTokenName(tokenName: string): xrpl.IssuedCurrency {
+    if (!coins.has(tokenName)) {
+      throw new UnsupportedTokenError(`${tokenName} is not supported`);
+    }
+    const token = coins.get(tokenName);
+    if (!token.isToken || !(token instanceof XrpCoin)) {
+      throw new UnsupportedTokenError(`${tokenName} is not an XRP token`);
```
Comment (dpkjnr) [inline modules/sdk-coin-xrp/src/lib/utils.ts:245]: Token might be a correct but not supported by Bitgo. message looks misleading

**Review by dpkjnr — APPROVED** [review]

**Pull request merged**